src/plugins/mvcc_delete_plugin.cpp

```diff
@@ -0,0 +1,171 @@
+  std::unique_lock<std::mutex> lock(_mutex_physical_delete_queue);
+
+  if (_physical_delete_queue.size()) {
+    TableAndChunkID table_and_chunk_id = _physical_delete_queue.front();
+    const auto& table = table_and_chunk_id.first;
+    const auto& chunk = table->get_chunk(table_and_chunk_id.second);
+
+    DebugAssert(chunk != nullptr, "Chunk does not exist. Physical Delete can not be applied.");
+
+    if (chunk->get_cleanup_commit_id().has_value()) {
+      // Check whether there are still active transactions that might use the chunk
+      bool conflicting_transactions = false;
+      auto lowest_snapshot_commit_id = Hyrise::get().transaction_manager.get_lowest_active_snapshot_commit_id();
+
+      if (lowest_snapshot_commit_id.has_value()) {
+        conflicting_transactions = chunk->get_cleanup_commit_id().value() > lowest_snapshot_commit_id.value();
+      }
+
+      if (!conflicting_transactions) {
+        _delete_chunk_physically(table, table_and_chunk_id.second);
+        _physical_delete_queue.pop();
+      }
+    }
+  }
+}
+
+bool MvccDeletePlugin::_try_logical_delete(const std::string& table_name, const ChunkID chunk_id) {
+  const auto& table = Hyrise::get().storage_manager.get_table(table_name);
+  const auto& chunk = table->get_chunk(chunk_id);
+
+  Assert(chunk != nullptr, "Chunk does not exist. Logical Delete can not be applied.");
+  Assert(chunk_id < (table->chunk_count() - 1),
+         "MVCC Logical Delete should not be applied on the last/current mutable chunk.");
+
+  // Create temporary referencing table that contains the given chunk only
+  //   Include all ChunksIDs of current table except chunk_id for pruning in GetTable
+  std::vector<ChunkID> excluded_chunk_ids(table->chunk_count() - 1);
+  std::iota(excluded_chunk_ids.begin(), excluded_chunk_ids.begin() + chunk_id, 0);
+  std::iota(excluded_chunk_ids.begin() + chunk_id, excluded_chunk_ids.end(), chunk_id + 1);
+
+  auto transaction_context = Hyrise::get().transaction_manager.new_transaction_context();
+
+  auto get_table = std::make_shared<GetTable>(table_name, excluded_chunk_ids, std::vector<ColumnID>());
+  get_table->set_transaction_context(transaction_context);
+  get_table->execute();
+
+  // Validate temporary table
+  auto validate = std::make_shared<Validate>(get_table);
+  validate->set_transaction_context(transaction_context);
+  validate->execute();
+
+  // Use Update operator to delete and re-insert valid records in chunk
+  // Pass validate into Update operator twice since data will not be changed.
+  auto update = std::make_shared<Update>(table_name, validate, validate);
+  update->set_transaction_context(transaction_context);
+  update->execute();
+
+  // Check for success
+  if (update->execute_failed()) {
+    // Transaction conflict. Usually, the OperatorTask would call rollback, but as we executed Update directly, that is
+    // our job.
+    transaction_context->rollback();
+    return false;
+  }
```

Comment:
I couldn't come up with a good way to test this. Any ideas?

Reply:
Ouch! Was not on my radar that we need an OperatorTask for Auto-Rollback in case of failures. 

With the current implementation, there is no way for testing transaction conflicts. 
An interface change might help though: We should create the transaction context outside of `_try_logical_delete(..)` and pass it in. 
I'll try that.

Reply:
Done, works!

Reply:
Good idea, thanks :)